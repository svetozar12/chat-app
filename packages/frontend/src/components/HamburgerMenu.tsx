import React from "react";
import { useSelector, useDispatch } from "react-redux";
import { InitialState2 } from "../redux/state";
import { GiHamburgerMenu } from "react-icons/gi";
function HamburgerMenu() {
  const dispatch = useDispatch();
  const state = useSelector(
    (state: { setReducer: InitialState2 }) => state.setReducer,
  );
  return (
    <GiHamburgerMenu
      className="hamburger"
      onClick={() =>
        dispatch({ type: "SET_MOBILE_NAV", payload: !state.setMobileNav })
      }
    />
  );
}

export default HamburgerMenu;
