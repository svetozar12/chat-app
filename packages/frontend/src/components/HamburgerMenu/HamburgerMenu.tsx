import { css, cx } from "@emotion/css";
import { useSelector, useDispatch } from "react-redux";
import { IInitialSet } from "../../services/redux/reducer/setReducer/state";
import { GiHamburgerMenu } from "react-icons/gi";
function HamburgerMenu() {
  const dispatch = useDispatch();
  const state = useSelector((state: { setReducer: IInitialSet }) => state.setReducer);

  return (
    <div
      className={cx(
        "flex",
        css`
          position: absolute;
          top: 0;
          left: 0;
          width: 3rem;
          height: 3rem;
          border: 1px solid transparent;
          margin: 1rem;
          &:active {
            background: var(--gradient-first);
            border-radius: 100%;
          }
        `,
      )}
    >
      <GiHamburgerMenu
        title="hamburger"
        className={css`
          display: inline-block;
          cursor: pointer;
          width: 2rem;
          height: 2rem;
          zindex: -11;
          @media (min-width: 980px) {
            display: none;
          } ;
        `}
        onClick={() => dispatch({ type: "SET_MOBILE_NAV", payload: !state.setMobileNav })}
      />
    </div>
  );
}

export default HamburgerMenu;
