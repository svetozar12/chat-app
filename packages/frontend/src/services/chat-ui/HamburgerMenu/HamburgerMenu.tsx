// eslint-disable-next-line import/named
import { Box, BoxProps } from '@chakra-ui/react';
import { css, cx } from '@emotion/css';
import { GiHamburgerMenu } from 'react-icons/gi';
import { IBaseComponent } from '../types';

type Base = IBaseComponent<BoxProps>;
interface IHambugerMenu extends Base {
  toggleHamburger: () => void;
}

function HamburgerMenu(props: IHambugerMenu) {
  const { toggleHamburger, baseProps, chakraProps, style } = props;
  return (
    <Box
      className={cx(
        'flex',
        css`
          position: absolute;
          top: 0;
          left: 0;
          width: 3rem;
          height: 3rem;
          border: 1px solid transparent;
          margin: 1rem;
          &:active {
            background: var(--gradient-first);
            border-radius: 100%;
          }
        `,
      )}
      {...chakraProps}
      {...style}
      {...baseProps}
    >
      <GiHamburgerMenu
        title="hamburger"
        className={css`
          display: inline-block;
          cursor: pointer;
          width: 2rem;
          height: 2rem;
          zindex: -11;
          @media (min-width: 980px) {
            display: none;
          } ;
        `}
        onClick={toggleHamburger}
      />
    </Box>
  );
}

export default HamburgerMenu;
