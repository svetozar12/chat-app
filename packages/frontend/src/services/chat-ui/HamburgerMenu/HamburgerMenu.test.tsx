import { Box } from '@chakra-ui/react';
import { mount, ReactWrapper } from 'enzyme';
import HamburgerMenu from 'services/chat-ui/HamburgerMenu/HamburgerMenu';

const componentProps = {
  toggleHamburger: jest.fn(),
  style: { background: 'lime' },
  chakraProps: { h: '20%' },
};

const baseProps = {
  background: 'lime',
  h: '20%',
};

const setupMount = () => {
  return mount(<HamburgerMenu {...componentProps} />);
};

describe('testing correct behaviour of <HamburgerMenu />', () => {
  let wrapper: ReactWrapper;
  beforeAll(() => {
    wrapper = setupMount();
  });
  it('component <HamburgerMenu /> should render', () => {
    expect(wrapper.find(HamburgerMenu).length).toBe(1);
  });
  Object.entries(baseProps).forEach(([key, value]) => {
    it(`component <HamburgerMenu /> should have prop ${key}:${value}`, () => {
      expect(wrapper.find(Box).prop(key)).toBe(value);
    });
  });
  it('component <HamburgerMenu /> should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
