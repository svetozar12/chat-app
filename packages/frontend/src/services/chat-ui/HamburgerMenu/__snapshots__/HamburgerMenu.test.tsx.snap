// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing correct behaviour of <HamburgerMenu /> component <HamburgerMenu /> should match snapshot 1`] = `
<HamburgerMenu
  chakraProps={
    Object {
      "h": "20%",
    }
  }
  style={
    Object {
      "background": "lime",
    }
  }
  toggleHamburger={[MockFunction]}
>
  <Box
    background="lime"
    className="flex css-npgs22"
    h="20%"
  >
    <Noop />
    <div
      className="flex css-npgs22 css-0"
    >
      <GiHamburgerMenu
        className="css-snhyxz"
        onClick={[MockFunction]}
        title="hamburger"
      >
        <IconBase
          attr={
            Object {
              "viewBox": "0 0 512 512",
            }
          }
          className="css-snhyxz"
          onClick={[MockFunction]}
          title="hamburger"
        >
          <svg
            className="css-snhyxz"
            fill="currentColor"
            height="1em"
            onClick={[MockFunction]}
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <title>
              hamburger
            </title>
            <path
              d="M32 96v64h448V96H32zm0 128v64h448v-64H32zm0 128v64h448v-64H32z"
              key="0"
            />
          </svg>
        </IconBase>
      </GiHamburgerMenu>
    </div>
  </Box>
</HamburgerMenu>
`;
